### PR TITLE
To Do list: List structure

### PR DESCRIPTION
**Pull request by Ali-0111**
**Objectives**

1. Using webpack to bundle JavaScript.
2. Learn how to use proper ES6 syntax.
3. Use ES6 modules to write modular JavaScript.

**Description**
In this project The list will be styled according to the specifications listed  in this lesson. This simple web page will be built using webpack and served by a webpack dev server.

### CODE REVIEW

File: src/modules/todoList.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+class List {
```
Comment (gemmen29) [inline src/modules/todoList.js:1]: - [ ] Kindly add some static data(array of objects) as per this [requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m1_list_structure.md#:~:text=set%20an%20array,index%3A%20%5Bnumber%5D.). You don't need to implement add todo function as it's upcoming in a later milestone

File: src/modules/todoList.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  show() {
+    this.items.innerHTML = this.db.map((element) => (`
+      <li class = "todo">
+        <button class="check"></button>
+        <p class="item-text">${element.description}</p>
+        <div class="moreBtn"></div>
+      </li>`
+    )).join('');
+  }
+}
```
Comment (gemmen29) [inline src/modules/todoList.js:40]: - [ ] Kindly sort the list of items before displaying it as per this [requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m1_list_structure.md#:~:text=The%20list%20should%20appear%20in%20order%20of%20the%20index%20values%20for%20each%20task.). The list should appear in order of the index values for each task.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @Ali-0111, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  },
+  {
+    description: 'Coding with Partner',
+    completed: false,
+    index: 3,
+  },
+];
+
+function show() {
+  items.innerHTML = data.map((element) => (`
```
Comment (OyePriscilla) [inline src/index.js:25]: - [ ] Kindly sort the list of items before displaying it as it was pointed out earlier according to the requirement. 👍 I think you can reference this [tutorial](https://www.tutorialspoint.com/sort-by-index-of-an-array-in-javascript#:~:text=const%20arr%20%3D%20%5B%20%7B%20'name,takes%20in%20one%20such%20array.).

You may try this as well:   `data.sort((a, b) => a.index - b.index)`

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @Ali-0111 :clap, 

Great work on making the changes requested by a previous reviewer 👏 

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close-so-close-seth-meyers

 ## Highlights
Use webpack to setup project ✔️ 
Implement an array of some simple to do tasks ✔️ 
Implement function that iterates over the tasks array ✔️ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by PinkMoon25 — APPROVED** [review]
Hi @Ali-0111 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Linters :heavy_check_mark: 
Github flow :heavy_check_mark: 
Webpack :heavy_check_mark: 

## Optional suggestions

**- [ ] [OPTIONAL] Great job rendering List structure :clap: , However please notice that currently your MIT link is broken you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish.**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**